### ISSUE TITLE
memory leakage somewhere

### ISSUE THREAD

**Issue by yegor256** [zerocracy/farm#360]
```
2017-12-11T17:22:25.945380+00:00 heroku[web.1]: Process running mem=609M(111.3%)
2017-12-11T17:22:25.945658+00:00 heroku[web.1]: Error R14 (Memory quota exceeded)
2017-12-11T17:22:45.993000+00:00 heroku[web.1]: Process running mem=609M(111.3%)
2017-12-11T17:22:45.993196+00:00 heroku[web.1]: Error R14 (Memory quota exceeded)
```

**Comment by 0crat** [comment]
The job [#360](https://github.com/zerocracy/farm/issues/360) assigned to @caarlos0[/z](https://www.0crat.com/u/caarlos0). The budget is 30 minutes, see [§4](http://datum.zerocracy.com/pages/policy.html#4). Please, read [§8](http://datum.zerocracy.com/pages/policy.html#8) and [§9](http://datum.zerocracy.com/pages/policy.html#9). If the task is not clear, read [this](http://www.yegor256.com/2015/02/16/it-is-not-a-school.html) and [this](http://www.yegor256.com/2015/01/15/how-to-cut-corners.html).

**Comment by 0crat** [comment]
The user @caarlos0[/z](https://www.0crat.com/u/caarlos0) resigned from [#360](https://github.com/zerocracy/farm/issues/360), please stop working

**Comment by 0crat** [comment]
The job [#360](https://github.com/zerocracy/farm/issues/360) assigned to @t-izbassar[/z](https://www.0crat.com/u/t-izbassar), here is [why](https://www.0crat.com/footprint/C3NDPUA8L/610867028). The budget is 30 minutes, see [§4](http://www.zerocracy.com/policy.html#4). Please, read [§8](http://www.zerocracy.com/policy.html#8) and [§9](http://www.zerocracy.com/policy.html#9). If the task is not clear, read [this](http://www.yegor256.com/2015/02/16/it-is-not-a-school.html) and [this](http://www.yegor256.com/2015/01/15/how-to-cut-corners.html).

**Comment by t-izbassar** [comment]
@g4s8 can you try to download this heapdump? As in your #369 pr you made it possible only to you as a user. And publish it somewhere accessible?

**Comment by g4s8** [comment]
@t-izbassar at moment it's not possible, it's blocking by this bug https://github.com/zerocracy/farm/issues/400

**Comment by t-izbassar** [comment]
@0crat waiting for #400

**Comment by 0crat** [comment]
> @0crat waiting for #400  ([here](https://github.com/zerocracy/farm/issues/360#issuecomment-369915969))

@t-izbassar The impediment for [#360](https://github.com/zerocracy/farm/issues/360) was registered successfully by @t-izbassar[/z](https://www.0crat.com/u/t-izbassar)

**Comment by t-izbassar** [comment]
@0crat refuse

**Comment by 0crat** [comment]
> @0crat refuse ([here](https://github.com/zerocracy/farm/issues/360#issuecomment-388640604))

@t-izbassar The user @t-izbassar[/z](https://www.0crat.com/u/t-izbassar) resigned from [#360](https://github.com/zerocracy/farm/issues/360), please stop working. Reason for job resignation: Order was cancelled

**Comment by 0crat** [comment]
Tasks refusal is discouraged, see [§6](http://www.zerocracy.com/policy.html#6): -15 point(s) just awarded to @t-izbassar[/z](https://www.0crat.com/u/t-izbassar)

**Comment by 0crat** [comment]
The job [#360](https://github.com/zerocracy/farm/issues/360) assigned to @krzyk[/z](https://www.0crat.com/u/krzyk), here is [why](https://www.0crat.com/footprint/C3NDPUA8L/1331717266); the budget is 30 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§8](http://www.zerocracy.com/policy.html#8) and [§9](http://www.zerocracy.com/policy.html#9); if the task is not clear, read [this](http://www.yegor256.com/2015/02/16/it-is-not-a-school.html) and [this](http://www.yegor256.com/2015/01/15/how-to-cut-corners.html)

**Comment by krzyk** [comment]
@t-izbassar @g4s8 were you ever able to download the heapdump using `TkDump` (I know the plan is to put it in S3). I'm wondering is there a point in adding myself to the list and periodically trying to get data from heapdump endpoint?

**Comment by g4s8** [comment]
@krzyk I think we need a new solution here, we already have heap dump on out-of memory https://github.com/zerocracy/farm/pull/369/files#diff-5bd6b85c2d6fc987875f4bf82de2a15a but we can't access it. BTW we can download it with `heroku ps:exec`, but only @yegor256 can do it.

**Comment by krzyk** [comment]
@0crat refuse

**Comment by 0crat** [comment]
> @0crat refuse ([here](https://github.com/zerocracy/farm/issues/360#issuecomment-389189817))

@krzyk The user @krzyk[/z](https://www.0crat.com/u/krzyk) resigned from [#360](https://github.com/zerocracy/farm/issues/360), please stop working. Reason for job resignation: Order was cancelled

**Comment by 0crat** [comment]
Tasks refusal is discouraged, see [§6](http://www.zerocracy.com/policy.html#6): -15 point(s) just awarded to @krzyk[/z](https://www.0crat.com/u/krzyk)

**Comment by 0crat** [comment]
The job [#360](https://github.com/zerocracy/farm/issues/360) assigned to @amihaiemil[/z](https://www.0crat.com/u/amihaiemil), here is [why](https://www.0crat.com/footprint/C3NDPUA8L/1360857823); the budget is 30 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§8](http://www.zerocracy.com/policy.html#8) and [§9](http://www.zerocracy.com/policy.html#9); if the task is not clear, read [this](http://www.yegor256.com/2015/02/16/it-is-not-a-school.html) and [this](http://www.yegor256.com/2015/01/15/how-to-cut-corners.html)

**Comment by 0crat** [comment]
@amihaiemil[/z](https://www.0crat.com/u/amihaiemil) this job was assigned to you 5days ago. It will be taken away from you soon, unless you close it, see [§8](http://www.zerocracy.com/policy.html#8). Read [this](http://www.yegor256.com/2014/04/13/no-obligations-principle.html) and [this](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html), please.

**Comment by amihaiemil** [comment]
@0crat refuse

**Comment by amihaiemil** [comment]
@0crat refuse

**Comment by 0crat** [comment]
> @0crat refuse ([here](https://github.com/zerocracy/farm/issues/360#issuecomment-395548439))

@amihaiemil The user @amihaiemil[/z](https://www.0crat.com/u/amihaiemil) resigned from [#360](https://github.com/zerocracy/farm/issues/360), please stop working. Reason for job resignation: Order was cancelled

**Comment by 0crat** [comment]
> @0crat refuse ([here](https://github.com/zerocracy/farm/issues/360#issuecomment-398682957))

@amihaiemil Job `gh:zerocracy/farm#360` is not assigned, can't get performer

**Comment by 0crat** [comment]
Tasks refusal is discouraged, see [§6](http://www.zerocracy.com/policy.html#6): -15 point(s) just awarded to @amihaiemil[/z](https://www.0crat.com/u/amihaiemil)

**Comment by 0pdd** [comment]
@yegor256 2 puzzles [#1202](https://github.com/zerocracy/farm/issues/1202), [#1203](https://github.com/zerocracy/farm/issues/1203) are still not solved; solved: [#371](https://github.com/zerocracy/farm/issues/371), [#425](https://github.com/zerocracy/farm/issues/425), [#725](https://github.com/zerocracy/farm/issues/725).

**Comment by g4s8** [comment]
@0crat assign me

**Comment by 0crat** [comment]
There is an unrecoverable failure on my side. Please, submit it [here](https://github.com/zerocracy/farm/issues):

```
PID: 4@aa0fe21a-ccc8-4a0f-920f-f4c2aaf4a20f, thread: pool-173-thread-1
com.zerocracy.farm.sync.SyncProject[77] java.lang.IllegalStateException: Failed to acquire "people.xml" in "PMO" in 2min: 6cc77ffce735/22min/0/2/true/false by zerocrat_bot Telegram Executor
```

[0.22.25](https://github.com/zerocracy/farm): Issue: zerocracy/farm#360, Comment: 399089419

**Comment by 0crat** [comment]
There is an unrecoverable failure on my side. Please, submit it [here](https://github.com/zerocracy/farm/issues):

```
PID: 4@aa0fe21a-ccc8-4a0f-920f-f4c2aaf4a20f, thread: pool-173-thread-1
com.zerocracy.radars.github.ReSafe[82] java.io.IOException: java.lang.IllegalStateException: Failed to acquire "people.xml" in "PMO" in 2min: 6cc77ffce735/22min/0/2/true/false by zerocrat_bot Telegram Executor
com.zerocracy.farm.sync.SyncProject[77] java.lang.IllegalStateException: Failed to acquire "people.xml" in "PMO" in 2min: 6cc77ffce735/22min/0/2/true/false by zerocrat_bot Telegram Executor
```

[0.22.25](https://github.com/zerocracy/farm): Issue: zerocracy/farm#360, Action: `created`

**Comment by 0pdd** [comment]
@yegor256 the puzzle [#1202](https://github.com/zerocracy/farm/issues/1202) is still not solved; solved: [#1203](https://github.com/zerocracy/farm/issues/1203), [#371](https://github.com/zerocracy/farm/issues/371), [#425](https://github.com/zerocracy/farm/issues/425), [#725](https://github.com/zerocracy/farm/issues/725).

**Comment by g4s8** [comment]
@0crat assign me

**Comment by 0crat** [comment]
> @0crat assign me ([here](https://github.com/zerocracy/farm/issues/360#issuecomment-399098863))

@g4s8 The job [#360](https://github.com/zerocracy/farm/issues/360) assigned to @g4s8[/z](https://www.0crat.com/u/g4s8), here is [why](https://www.0crat.com/footprint/C3NDPUA8L/1721313034); the budget is 30 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§8](http://www.zerocracy.com/policy.html#8) and [§9](http://www.zerocracy.com/policy.html#9); if the task is not clear, read [this](http://www.yegor256.com/2015/02/16/it-is-not-a-school.html) and [this](http://www.yegor256.com/2015/01/15/how-to-cut-corners.html); we should be aware that @g4s8[/z](https://www.0crat.com/u/g4s8) is on vacation; this ticket may be delayed; there will be a monetary reward for this job

**Comment by 0crat** [comment]
Manual assignment of issues is discouraged, see [§19](http://www.zerocracy.com/policy.html#19): -5 point(s) just awarded to @g4s8[/z](https://www.0crat.com/u/g4s8)

**Comment by 0pdd** [comment]
@yegor256 all 5 puzzles are solved here: [#1202](https://github.com/zerocracy/farm/issues/1202), [#1203](https://github.com/zerocracy/farm/issues/1203), [#371](https://github.com/zerocracy/farm/issues/371), [#425](https://github.com/zerocracy/farm/issues/425), [#725](https://github.com/zerocracy/farm/issues/725).

**Comment by g4s8** [comment]
@rultor release, tag=`0.22.30`

**Comment by g4s8** [comment]
@yegor256 it should be partially fixed in #1408 - now we are using soft references for caching datum documents instead of static fields.

**Comment by yegor256** [comment]
@g4s8 OK, thanks
